test/test_type_promotion.py

```diff
@@ -312,6 +312,64 @@ def test_lt_with_type_promotion(self):
             self.assertTrue(actual, expected)
             self.assertTrue(actual.dtype == torch.bool)
 
+    def test_sparse(self):
+        t = torch.full([5, 5], 5, dtype=torch.int, device=self.device)
+        t[0, 0] = 0
+        t[1, 1] = 0
+        t[2, 2] = 0
+        s = t.to_sparse()
+        self.assertEqual((s + s).to_dense(), t + t)
+        self.assertEqual(t + s, (s + s).to_dense())
+        self.assertEqual(t + s.to(torch.double), (t + t))
+        self.assertEqual((t + s.to(torch.double)).dtype, torch.double)
+        self.assertEqual((s + s.to(torch.double)).to_dense(), t + t)
+        self.assertEqual((s + s.to(torch.double)).dtype, torch.double)
+
+        inplace = s.to(torch.float, copy=True)
```

Comment:
There are a lot of test cases here. I imagine when we add type promotion to more sparse ops, we will not want to just append LOC to this test case (`test_sparse`). Can we split this up into semantically meaningful tests for different ops and different overloads?